src/MICore/Transports/LocalLinuxTransport.cs

```diff
@@ -80,7 +80,19 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
                 throw new Exception(MICoreResources.Error_InvalidMiDebuggerPath);
             }
 
-            string debuggeeDir = System.IO.Path.GetDirectoryName(options.ExePath);
+            // Default working directory is next to the app
+            string debuggeeDir;
+            if (Path.IsPathRooted(options.ExePath) && File.Exists(options.ExePath))
+            {
+                debuggeeDir = System.IO.Path.GetDirectoryName(options.ExePath);
+            }
+            else
+            {
+                // If we don't know where the app is, default to HOME, and if we somehow can't get that, go with the root directory.
+                debuggeeDir = Environment.GetEnvironmentVariable("HOME");
+                if (string.IsNullOrEmpty(debuggeeDir))
+                    debuggeeDir = "/";
+            }
```

Comment:
I'll file a bug in the morning to track this

Sent from my Windows Phone
________________________________
From: Gregg Miskelly<mailto:notifications@github.com>
Sent: ‎3/‎17/‎2016 9:15 PM
To: Microsoft/MIEngine<mailto:MIEngine@noreply.github.com>
Cc: Jackson Davis<mailto:Jackson.Davis@microsoft.com>
Subject: Re: [MIEngine] Add support for non-absolute paths in 'ExePath' (#228)


In src/MICore/Transports/LocalLinuxTransport.cs<https://github.com/Microsoft/MIEngine/pull/228#discussion_r56611675>:

> @@ -80,7 +80,19 @@ public override void InitStreams(LaunchOptions options, out StreamReader reader,
>                  throw new Exception(MICoreResources.Error_InvalidMiDebuggerPath);
>              }
>
> -            string debuggeeDir = System.IO.Path.GetDirectoryName(options.ExePath);
> +            // Default working directory is next to the app
> +            string debuggeeDir;
> +            if (Path.IsPathRooted(options.ExePath) && File.Exists(options.ExePath))
> +            {
> +                debuggeeDir = System.IO.Path.GetDirectoryName(options.ExePath);
> +            }
> +            else
> +            {
> +                // If we don't know where the app is, default to HOME, and if we somehow can't get that, go with the root directory.
> +                debuggeeDir = Environment.GetEnvironmentVariable("HOME");
> +                if (string.IsNullOrEmpty(debuggeeDir))
> +                    debuggeeDir = "/";
> +            }


Yeah, this code seems a bit weird to me too. That option setting will work because we issue a '-environment-cd' to GDB to set the working directory. This just establishes kind of an initial working directory. I think honestly this code should probably be removed and replaced with something (see debuggedprocess.cs(556)). But I didn't have a good way of proving out all the cases, so I was hesitant to just remove it. So I just wanted to avoid breaking what was here.

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/MIEngine/pull/228/files/33fed37f2a2d3a500335ba1e66ab1d2f58d5ba22#r56611675>